src/it/java/teammates/it/sqllogic/api/EmailGeneratorTestIT.java

```diff
@@ -0,0 +1,143 @@
+package teammates.it.sqllogic.api;
+
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.SqlDataBundle;
+import teammates.common.util.Config;
+import teammates.common.util.EmailType;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.HibernateUtil;
+import teammates.common.util.TimeHelper;
+import teammates.it.test.BaseTestCaseWithSqlDatabaseAccess;
+import teammates.sqllogic.api.SqlEmailGenerator;
+import teammates.storage.sqlentity.FeedbackSession;
+import teammates.storage.sqlentity.Student;
+import teammates.test.EmailChecker;
+
+/**
+ * SUT: {@link SqlEmailGenerator}.
+ */
+public class EmailGeneratorTestIT extends BaseTestCaseWithSqlDatabaseAccess {
+
+    private final SqlEmailGenerator emailGenerator = SqlEmailGenerator.inst();
+
+    private SqlDataBundle dataBundle;
+
+    @Override
+    @BeforeClass
+    public void setupClass() {
+        super.setupClass();
+    }
+
+    @Override
+    @BeforeMethod
+    public void setUp() throws Exception {
+        super.setUp();
+        dataBundle = loadSqlDataBundle("/SqlEmailGeneratorTest.json");
+
+        FeedbackSession session1InCourse3 = dataBundle.feedbackSessions.get("session1InCourse3");
+        FeedbackSession session2InCourse3 = dataBundle.feedbackSessions.get("session2InCourse3");
+        FeedbackSession session1InCourse4 = dataBundle.feedbackSessions.get("session1InCourse4");
+        FeedbackSession session2InCourse4 = dataBundle.feedbackSessions.get("session2InCourse4");
+        // opened and unpublished.
+        session1InCourse3.setStartTime(TimeHelper.getInstantDaysOffsetFromNow(-20));
+        dataBundle.feedbackSessions.put("session1InCourse3", session1InCourse3);
+
+        // closed and unpublished
+        session2InCourse3.setStartTime(TimeHelper.getInstantDaysOffsetFromNow(-19));
+        session2InCourse3.setEndTime(TimeHelper.getInstantDaysOffsetFromNow(-1));
+        // session2InCourse3.setDeletedTime(null);
```

Comment:
nit: remove this comment if not needed

Reply:
Resolved should I get another reviewer or should this be merged?